shark-graph/src/test/java/shark/LongScatterSetTest.kt

```diff
@@ -1,142 +1,137 @@
+    val set = LongScatterSet()
+
+    set += SAME_MIX_PHI_1
+    set += SAME_MIX_PHI_2
 
-    // Verify that calling ensureCapacity didn't change any data in set
     assertThat(set)
-        .contains(10, 42)
+        .contains(SAME_MIX_PHI_1, SAME_MIX_PHI_2)
         .hasSize(2)
   }
 
-  private fun runAssertionsForRemoveOperation(set: LongScatterSet) {
-    require(set.size() == 0) { "Set should be empty in order to run the assertions on it" }
+  @Test fun `removing elements`() {
+    val set = LongScatterSet()
+    TEST_VALUES_LIST.forEach { set.add(it) }
 
-    val testValues = listOf(42, 0, Long.MIN_VALUE, Long.MAX_VALUE, -1)
+    TEST_VALUES_LIST.forEach { set.remove(it) }
 
-    // First, verify removing unique values from an empty set
-    testValues.forEach { value ->
-      set.remove(value)
-      assertThat(set)
-          .isEmpty()
-          .doesNotContain(value)
-    }
+    assertThat(set)
+        .doesNotContain(*TEST_VALUES_LIST.toLongArray())
+        .isEmpty()
+  }
 
-    // Add test values to the set
-    testValues.forEach { set += it }
+  @Test fun `removing from empty set`() {
+    val set = LongScatterSet()
 
-    // Verify removing unique values from non-empty set
-    testValues.forEachIndexed { index: Int, value: Long ->
-      // Value should be in the set at first
-      assertThat(set).contains(value)
+    set.remove(TEST_VALUE)
 
-      // Now remove the value from set
-      set.remove(value)
+    assertThat(set)
+        .doesNotContain(TEST_VALUE)
+        .isEmpty()
+  }
 
-      // Value is removed -> size should decrease, other elements should still be in set
-      assertThat(set)
-          .doesNotContain(value)
-          .hasSize(testValues.size - index - 1)
-      for (i in index + 1 until testValues.size) {
-        assertThat(set).contains(testValues[i])
-      }
-    }
+  @Test fun `removing elements with matching hash`() {
+    val set = LongScatterSet()
+    set.add(SAME_MIX_PHI_1)
+    set.add(SAME_MIX_PHI_2)
 
-    // Verify removing same element twice works correctly.
-    set.add(42)
-    set.remove(42)
-    set.remove(42)
+    set.remove(SAME_MIX_PHI_2)
+    set.remove(SAME_MIX_PHI_1)
 
     assertThat(set)
+        .doesNotContain(SAME_MIX_PHI_1, SAME_MIX_PHI_2)
         .isEmpty()
-        .doesNotContain(42)
+  }
+
+  @Test fun `release operation cleans set`() {
+    val set = LongScatterSet()
+    set.add(TEST_VALUE)
 
-    // Verify removing elements with matching hash. Remove in reverse order than inserting
-    set += 11
-    set += 14723950898
+    set.release()
 
-    set.remove(14723950898)
-    set.remove(11)
     assertThat(set)
+        .doesNotContain(TEST_VALUE)
         .isEmpty()
-        .doesNotContain(11, 14723950898)
   }
 
-  private fun runAssertionsForAddOperation(set: LongScatterSet) {
-    require(set.size() == 0) { "Set should be empty in order to run the assertions on it" }
+  @Test fun `setting initial capacity before operations`() {
```

Comment:
What is this testing exactly? Not obvious at all from test name and test content.